frontend/src/bundles/common/components/user-photo-uploader/user-photo-uploader.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback } from 'react';
+import { IconContext } from 'react-icons';
+import { GrGallery } from 'react-icons/gr';
+
+import styles from './styles.module.scss';
+
+interface UploadProperties {
+    setImage: React.Dispatch<React.SetStateAction<string | ArrayBuffer | null>>
+}
+
+const UserPhotoUploader: React.FC<UploadProperties> = ({ setImage }) => {
+
+    const handleFile = useCallback((file: File) => {
+        const reader = new FileReader();
+        reader.addEventListener('load', () => {
+            if (reader.readyState === 2) {
+                setImage(reader.result);
+            }
+        });
+        reader.readAsDataURL(file);
+    }, [setImage]);
+  
+    const handleFileChange  = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files?.[0];
+        if (file) {
+            handleFile(file);
```

Comment:
Shouldn't we have some restrictions for uploaded file (format, size)?

Reply:
Done